File: Base 7 - Leetcode 504/Base 7 - Leetcode 504.cpp
#include <string>
#include <algorithm>

class Solution {
public:
    std::string convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        int originalNum = num;
        num = abs(num);
        std::string remainders;

        while (num > 0) {
            int remainder = num % 7;
            remainders.push_back(remainder + '0');
            num /= 7;
        }

        if (originalNum < 0) {
            remainders.push_back('-');
        }

        std::reverse(remainders.begin(), remainders.end());
        return remainders;
    }
};

File: Base 7 - Leetcode 504/Base 7 - Leetcode 504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return '0'

        original_num = num
        num = abs(num)
        remainders = []
        
        while num > 0:
            remainder = num % 7
            remainders.append(str(remainder))
            num //= 7
        
        if original_num < 0:
            remainders.append('-')
        remainders.reverse()
        return ''.join(remainders)
        
        # Time and Space: O(log_7(N))